core/services/mavftp/mavftp.py

```diff
@@ -0,0 +1,680 @@
+# As this file is basically copied over from https://github.com/ArduPilot/MAVProxy/blob/master/MAVProxy/modules/mavproxy_ftp.py
```

Comment:
Can't we just install it as a module with a bootstrap script ? Over duplicating the code.

Reply:
the original code doesn't really have the functionality we need. but as the goal is to eventually move it up to ardupiot, perhaps I could keep it in a different repo and use a bootstrap file. thoughts?

Reply:
wait, but is this file the original code or not ? what was changed ? I believe that the best way to track it is to use bootstrap and a branch of ardupilot with the changes that you want to apply on their remote, so we can track the changes and also the PR.

Reply:
this file is derived from the original code. A lot of the original code is gone, it is not organized in classes, linted, typed, and provides a beter interface to use.

I'm not sure this should live in the main ardupilot repo at all. maybe at pymavlink/tools? I think that makes sense